Core/Types/GameComparator/StrictGameComparator.cs

```diff
@@ -51,8 +52,24 @@ public bool Compatible(KspVersion gameVersion, CkanModule module)
             {
                 return true;
             }
+            
+            if (!moduleRange.Upper.Value.IsAny && isBoundLower(moduleRange.Upper, gameVersionRange.Lower))
+            {
+                return false;
+            }
+
+            if (!moduleRange.Lower.Value.IsAny && isBoundLower(gameVersionRange.Upper, moduleRange.Lower))
+            {
+                return false;
+            }
+
+            return true;
+        }
 
-            return moduleRange.IsSupersetOf(gameVersionRange);
```

Comment:
Should be fixed in 0763a89, the problem was unbounded bounds (i.e. `KspVersionBound.Unbounded`) needed to be treated as highest and lowest possible values in `KspVersionBound.Highest()` and `KspVersionBound.Lowest()`. Test cases have been added for those situations. Thanks.